kaspresso/src/main/kotlin/com/kaspersky/kaspresso/docloc/metadata/extractor/UiMetadataExtractor.kt

```diff
@@ -0,0 +1,96 @@
+package com.kaspersky.kaspresso.docloc.metadata.extractor
+
+import android.app.Activity
+import androidx.test.uiautomator.By
+import androidx.test.uiautomator.StaleObjectException
+import androidx.test.uiautomator.UiDevice
+import com.kaspersky.kaspresso.device.activities.Activities
+import com.kaspersky.kaspresso.docloc.metadata.LocalizedString
+import com.kaspersky.kaspresso.docloc.metadata.Metadata
+import com.kaspersky.kaspresso.docloc.metadata.Window
+import com.kaspersky.kaspresso.logger.UiTestLogger
+
+internal class UiMetadataExtractor(
+    private val uiDevice: UiDevice,
+    private val activities: Activities,
+    private val logger: UiTestLogger,
+) : MetadataExtractor {
+
+    private val metadataExtractorHelper = MetadataExtractorHelper()
+
+    override fun getMetadata(): Metadata {
+        val activity = activities.getResumed() ?: throw RuntimeException("Failed to get current activity")
+        return createMetadata(activity)
+    }
+
+    private fun createMetadata(activity: Activity): Metadata {
+        val localizedStrings = metadataExtractorHelper.resolveAmbiguous(
+            getLocalizedStrings()
+        )
+        val window = activity.window.decorView.run {
+            Window(
+                left,
+                top,
+                width,
+                height,
+                localizedStrings
+            )
+        }
+        return Metadata(window)
+    }
+
+    private fun getLocalizedStrings(): List<LocalizedString> {
+        uiDevice.setCompressedLayoutHierarchy(false)
+        val objectsWithText = uiDevice.findObjects(By.enabled(true))
```

Comment:
100% needs better UiObjects tree parsing. Probably would be better if we used recursion